pkg/client/clientcmd/client_config.go

```diff
@@ -250,6 +250,9 @@ func (config DirectClientConfig) Namespace() (string, error) {
 	if len(configContext.Namespace) == 0 {
 		return api.NamespaceDefault, nil
 	}
+	if configContext.Namespace == "*" {
```

Comment:
> I don't know that we want to allow namespace to be stored in the config as *. That violates some of the fundamentals of our context settings.

It means only config file such as `$HOME/.kube/config` or both config file and argument? I assume the latter.



> Perhaps we can add a boolean 'selectAll' which is used if the namespace is empty?

IIUC, empty namespace is handled as _default_ namespace, can it change?
Or,  how about adding something like `--all-namespace` for argument and `allNamespace: true|false` option for config file to handle all namespace.